### PR TITLE
Ulfm/ishrink: add the MPI_COMM_ISHRINK operation

### PR DESCRIPTION
**Pull request by abouteiller**
Add the MPI_COMM_ISHRINK operation, as defined in more recent ULFM standard draft

### CODE REVIEW

File: ompi/mpiext/ftmpi/mpif-h/comm_ishrink_f.c [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+#include "ompi/mpiext/ftmpi/c/mpiext_ftmpi_c.h"
+#include "ompi/mpiext/ftmpi/mpif-h/prototypes_mpi.h"
+
+#include "ompi/communicator/communicator.h"
+#include "ompi/request/request.h"
+#include "ompi/mpi/fortran/base/fint_2_int.h"
+
+#if OPAL_HAVE_WEAK_SYMBOLS
+#pragma weak PMPIX_COMM_SHRINK = ompix_comm_shrink_f
```
Comment (bosilca) [inline ompi/mpiext/ftmpi/mpif-h/comm_ishrink_f.c:24]: copy/paste issue. use ishrink

**Comment by ibm-ompi** [comment]
The IBM CI (PGI) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/47321c6215b557fe47569d5f512da70a

File: ompi/mpiext/ftmpi/mpif-h/comm_ishrink_f.c (hunk 1 above)
Comment (abouteiller) [reply ompi/mpiext/ftmpi/mpif-h/comm_ishrink_f.c]: Thanks for pointing out, there was actually more things wrong with the Fortran bindings, it was missing a number of f08 functions etc. It's all good now.

**Review by jsquyres — COMMENTED** [review]
Can you add a man page, too?

**Comment by dalcinl** [comment]
@abouteiller For consistency with the changes in #10507, could you please remember adding the following line
```c
#define OMPI_HAVE_MPIX_COMM_ISHRINK 1
```
at the end of `ompi/mpiext/ftmpi/c/mpiext_ftmpi_c.h` ?

**Comment by abouteiller** [comment]
Needs to be rebased.

**Comment by bosilca** [comment]
Rebased and with the #define. Once CI validated this is ready to go.

**Review by bosilca — APPROVED** [review]

**Comment by dalcinl** [comment]
@abouteiller @bosilca I think you guys forgot to declare `PMPIX_Comm_shrink` in the header file.

```c
$ grep shrink /home/devel/mpi/openmpi/dev/include/openmpi/mpiext/mpiext_ftmpi_c.h 
OMPI_DECLSPEC int MPIX_Comm_shrink(MPI_Comm comm, MPI_Comm *newcomm);
OMPI_DECLSPEC int MPIX_Comm_ishrink(MPI_Comm comm, MPI_Comm *newcomm, MPI_Request *request);
OMPI_DECLSPEC int PMPIX_Comm_shrink(MPI_Comm comm, MPI_Comm *newcomm);
```

**Comment by abouteiller** [comment]
Later issue fixed with #11447

**Pull request merged**